api/src/pcapi/connectors/ems.py

```diff
@@ -1,73 +1,130 @@
+        if response.status_code >= 400:
+            reason = self._extract_reason_from_response(response)
+            message = self._extract_message_from_response(response)
+            raise EMSAPIException(f"Error on EMS API call {reason} - {message}")
 
+    def _extract_reason_from_response(self, response: requests.Response) -> str:
+        if isinstance(response.reason, bytes):
+            try:
+                reason = response.reason.decode("utf-8")
+            except UnicodeDecodeError:
+                reason = response.reason.decode("iso-8859-1")
+        else:
+            reason = response.reason
+        return reason
 
-def get_movie_poster_from_api(image_url: str) -> bytes:
-    api_response = requests.get(image_url)
+    def _extract_message_from_response(self, response: requests.Response) -> str:
+        try:
+            content = response.json()
+            message = content.get("message", "")
+        except json.JSONDecodeError:
+            message = response.content
+        return message
+
+
+class EMSBookingConnector:
+    digest_mode = "sha512"
+    booking_endpoint = "VENTE/"
+
+    def do_request(self, endpoint: str, payload: dict) -> Response:
+        """
+        Perform the actual request, using mandatory headers
+        and hashed payload.
+        """
+        headers = self._build_headers()
+        url = self._build_url(endpoint, payload)
+        return requests.post(url, headers=headers, json=payload)
+
+    def raise_for_status(self, response: Response) -> None:
+        response.raise_for_status()
+        content = response.json()
+        if content.get("statut") != 1:
+            raise EMSAPIException(f'Error on EMS API with {content["code_erreur"]} - {content["message_erreur"]}')
 
-    if api_response.status_code != 200:
-        raise EMSAPIException(f"Error getting EMS API movie poster {image_url} with code {api_response.status_code}")
+    def _build_headers(self) -> dict[str, str]:
+        """
+        Mandatory header to go through EMS API
+        """
+        return {"Source": settings.EMS_API_BOOKING_HEADER, "Content-Type": "application/json"}
 
-    return api_response.content
+    def _build_url(self, endpoint: str, payload: dict) -> str:
+        """
+        Build the url for EMS booking API according to their specifications.
 
+        The payload that is going to be sent is also hashed with sha512 and a secret key
+        and will be concatenate to the url we are calling.
 
-def _check_response_is_ok(response: requests.Response) -> None:
-    if response.status_code >= 400:
-        reason = _extract_reason_from_response(response)
-        message = _extract_message_from_response(response)
-        error_message = _filter_credentials(reason)
-        raise EMSAPIException(f"Error on EMS API call {error_message} - {message}")
+        Args:
+            endpoint (str): Endpoint we want to call (SEANCE/, VENTE/, ANNULATION/, ...)
```

Comment:
```suggestion
            endpoint (str): Endpoint we want to call (SEANCE, VENTE, ANNULATION, ...)
```
si on construit l'url comme suggéré plus bas